osf/utils/caching.py

```diff
@@ -61,6 +61,23 @@ def do_fdel(obj):
 
         return do_fdel
 
+class _CachedNotNoneProperty(_CachedProperty):
+
+    def _wrap_fget(self, fget):
+        @wraps(fget)
+        def do_fget(obj):
+            if hasattr(obj, self._cache_name):
+                if getattr(obj, self._cache_name):
+                    return getattr(obj, self._cache_name)
+            # Generate the value to cache.
+            value = fget(obj)
+            if value:
```

Comment:
Minor: `_CachedTruthyProperty` would be a more accurate name for this behavior, or if this conditional was `if value is not None:` the current name would be more accurate